ipa-core/src/protocol/ipa_prf/boolean_ops/addition_sequential.rs

```diff
@@ -25,20 +25,24 @@ use crate::{
 ///
 /// # Errors
 /// propagates errors from multiply
-pub async fn integer_add<C, F, XS, YS, const N: usize>(
+pub async fn integer_add<C, const N: usize>(
     ctx: C,
     record_id: RecordId,
-    x: &XS,
-    y: &YS,
-) -> Result<(XS, AdditiveShare<F, N>), Error>
+    x: &BitDecomposed<AdditiveShare<Boolean, N>>,
+    y: &BitDecomposed<AdditiveShare<Boolean, N>>,
+) -> Result<
+    (
+        BitDecomposed<AdditiveShare<Boolean, N>>,
+        AdditiveShare<Boolean, N>,
+    ),
+    Error,
+>
```

Comment:
Makes sense to me. This is the only way I can see us using this method. Nice to remove some of these extra generics =). Makes it easier to understand how it's used.